src/player.cpp

```diff
@@ -7640,6 +7643,90 @@ void player::suffer()
         hurtall(radiation / 100, nullptr);
     }
 
+    if (reactor_plut || tank_plut || slow_rad) {
+        // Microreactor CBM and supporting bionics
+        if (has_bionic("bio_reactor") || has_bionic("bio_advreactor")) {
+            //first do the filtering of plutonium from storage to reactor
+            int plut_trans;
+            plut_trans = 0;
+            if (tank_plut > 0) {
+                if (has_active_bionic("bio_plut_filter")) {
+                    plut_trans = (tank_plut * 0.025);
+                } else if (has_bionic("bio_plut_filter")) {
+                    plut_trans = (tank_plut * 0.01);
+                } else {
+                    plut_trans = (tank_plut * 0.005);
+                }
+                if (plut_trans < 1) {
+                    plut_trans = 1;
+                }
+                tank_plut -= plut_trans;
+                reactor_plut += plut_trans;
+            }
+            //leaking radiation, reactor is unshielded, but still better than a simple tank
+            slow_rad += ((tank_plut * 0.1) + (reactor_plut * 0.01));
+            int power_gen;
+            power_gen = 0;
+            //begin power generation
+            if (reactor_plut > 0) {
+                if (has_bionic("bio_advreactor")){
+                    if ((reactor_plut * 0.2) > 500){
+                        power_gen = 500;
+                    } else {
+                        power_gen = reactor_plut * 0.05;
+                    }
+                    slow_rad += (power_gen / 2); //upgraded reactor is less leaky
+                    while (slow_rad >= 100) {
+                        if (power_gen >= 1) {
+                            slow_rad -= 100;
+                            power_gen -= 1;
+                            reactor_plut -= 1;
+                        } else {
+                        break;
+                        }
+                    }
+                } else if (has_bionic("bio_reactor")) {
+                    if ((reactor_plut * 0.1) > 50){
+                        power_gen = 50;
+                    } else {
+                        power_gen = reactor_plut * 0.025;
+                    }
+                    slow_rad += (power_gen * 2); //lack of radiation shielding means that you get bonus radiation
+                }
+                reactor_plut -= power_gen;
+                if (power_gen > (max_power_level - power_level)) {
+                    power_gen -= (max_power_level - power_level);
+                    power_level = max_power_level;
+                    if (power_gen >= 50) { //more power than you can store is painful and harmful
```

Comment:
Should be more along the lines of power gen exceeding a certain amount per turn (haven't looked, and this'd need the other systems to be cross-checked), you overload your internal circuits and take damage/stun from that, IMO.